src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -217,79 +238,91 @@ private ImmutableArray<Document> GetPriorityDocuments(Project project)
             // rest of the results following soon after as best as we can find them.
             foreach (var projectGroup in orderedProjects)
             {
-                var allTasks = projectGroup.Select(p => Task.Run(() => SearchAsync(p, isFullyLoaded, seenItems, cancellationToken)));
+                var allTasks = projectGroup.Select(p => Task.Run(() => SearchCoreAsync(p)));
                 await Task.WhenAll(allTasks).ConfigureAwait(false);
             }
 
-            return seenItems.Count > 0;
-        }
+            return;
 
-        private async Task SearchAsync(
-            Project project, bool isFullyLoaded, HashSet<INavigateToSearchResult> seenItems, CancellationToken cancellationToken)
-        {
-            try
+            async Task SearchCoreAsync(Project project)
             {
-                await SearchCoreAsync(project, isFullyLoaded, seenItems, cancellationToken).ConfigureAwait(false);
-            }
-            finally
-            {
-                await _progress.ItemCompletedAsync(cancellationToken).ConfigureAwait(false);
+                try
+                {
+                    // If they don't even support the service, then always show them as having done the
+                    // complete search.  That way we don't call back into this project ever.
+                    var service = _host.GetNavigateToSearchService(project);
+                    if (service == null)
+                        return;
+
+                    await searchProjectAsync(service, project, result =>
+                    {
+                        // If we're seeing a dupe in another project, then filter it out here.  The results from
+                        // the individual projects will already contain the information about all the projects
+                        // leading to a better condensed view that doesn't look like it contains duplicate info.
+                        lock (seenItems)
+                        {
+                            if (!seenItems.Add(result))
+                                return Task.CompletedTask;
+                        }
+
+                        return _callback.AddItemAsync(project, result, cancellationToken);
+                    }).ConfigureAwait(false);
+                }
+                finally
+                {
+                    // after each project is searched, increment our progress.
+                    await _progress.ItemCompletedAsync(cancellationToken).ConfigureAwait(false);
+                }
             }
         }
 
-        private async Task SearchCoreAsync(
-            Project project, bool isFullyLoaded, HashSet<INavigateToSearchResult> seenItems, CancellationToken cancellationToken)
+        private Task SearchFullyLoadedProjectsAsync(
+            ImmutableArray<ImmutableArray<Project>> orderedProjects,
+            HashSet<INavigateToSearchResult> seenItems,
+            CancellationToken cancellationToken)
         {
```

Comment:
the next three search methods just defer to the helper above, simply passing in exactly what type of search they actually want to do on a project.